toolkit/tools/imagegen/configuration/partitionflag_test.go

```diff
@@ -0,0 +1,79 @@
+// Copyright Microsoft Corporation.
+// Licensed under the MIT License.
+
+package configuration
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+//TestMain found in configuration_test.go.
+
+var (
+	validPartitionFlags = []PartitionFlag{
+		PartitionFlag("esp"),
+		PartitionFlag("grub"),
+		PartitionFlag("bios_grub"),
+		PartitionFlag("boot"),
+		PartitionFlag("dmroot"),
+	}
+	invalidPartitionFlag     = PartitionFlag("not_a_partition_flag")
+	validPartitionFlagJSON   = `"esp"`
+	invalidPartitionFlagJSON = `1234`
+)
+
+func TestShouldSucceedValidFlagsMatch_PartitionFlag(t *testing.T) {
+	var pf PartitionFlag
+	assert.Equal(t, len(validPartitionFlags), len(pf.GetValidPartitionFlags()))
+
+	for _, partitionFlag := range validPartitionFlags {
+		found := false
+		for _, validPartitionType := range pf.GetValidPartitionFlags() {
+			if partitionFlag == validPartitionType {
+				found = true
+			}
+		}
+		assert.True(t, found)
+	}
+}
```

Comment:
I'm not sure I get this test case. Why use two separate arrays and compare their contents, if one would make this test redundant?

Reply:
In hindsight, would probably be better to explicitly test each constant individually. The goal is to make sure the constants are what we expect them to be. I'll switch to explicitly checking each one since that will be more resilient. (And go back and update some previous tests that do the same)

Reply:
Ok, having implemented the new way I actually like the old way better. It makes sure we have a solid understanding of the expected constants (as defined by `validPartitionFlags` in the test file).

It does this by making sure that our `PartitionFlag` struct returns the exact same number of constants, and that each constant is found at least once. Combining those two we can guarantee the sets are equivalent. This will catch any addition, removal, or modification of a constant. Since the `IsValid()` function uses the same `GetValidPartitionFlags()` to validate we have a strong guarantee we won't accidently break someone's config file.